src/sysfs_fuse.c

```diff
@@ -264,28 +466,49 @@ __lxcfs_fuse_ops int sys_open(const char *path, struct fuse_file_info *fi)
 	return 0;
 }
 
-__lxcfs_fuse_ops int sys_access(const char *path, int mask)
+__lxcfs_fuse_ops int sys_opendir(const char *path, struct fuse_file_info *fi)
 {
-	if (strcmp(path, "/sys") == 0 && access(path, R_OK) == 0)
-		return 0;
+	__do_free struct file_info *dir_info = NULL;
+	int type = -1;
 
-	if (strcmp(path, "/sys/devices") == 0 && access(path, R_OK) == 0)
-		return 0;
+	if (!liblxcfs_functional())
+		return -EIO;
 
-	if (strcmp(path, "/sys/devices/system") == 0 && access(path, R_OK) == 0)
-		return 0;
+	if (strcmp(path, "/sys") == 0)
+		type = LXC_TYPE_SYS;
+	if (strcmp(path, "/sys/devices") == 0)
+		type = LXC_TYPE_SYS_DEVICES;
+	if (strcmp(path, "/sys/devices/system") == 0)
+		type = LXC_TYPE_SYS_DEVICES_SYSTEM;
+	if (strcmp(path, "/sys/devices/system/cpu") == 0)
+		type = LXC_TYPE_SYS_DEVICES_SYSTEM_CPU;
+	if (strncmp(path, "/sys/devices/system/cpu/", STRLITERALLEN("/sys/devices/system/cpu/")) == 0 && S_ISDIR(get_st_mode(path)))  
+		type = LXC_TYPE_SYS_DEVICES_SYSTEM_CPU_SUBDIR;
+	if (type == -1)
+		return -ENOENT;
 
-	if (strcmp(path, "/sys/devices/system/cpu") == 0 &&
-	    access(path, R_OK) == 0)
```

Comment:
Where did the access check here go?

Reply:
I simply call libc's access(path, mask) for all kinds files , so i delete the access check here and read-only check.